packages/edit-post/src/components/layout/index.js

```diff
@@ -71,6 +71,53 @@ const interfaceLabels = {
 	footer: __( 'Editor footer' ),
 };
 
+function useEditorStyles() {
+	const { hasThemeStyleSupport, editorSettings } = useSelect(
+		( select ) => ( {
+			hasThemeStyleSupport:
+				select( editPostStore ).isFeatureActive( 'themeStyles' ),
+			editorSettings: select( editorStore ).getEditorSettings(),
+		} )
+	);
+
+	// Compute the default styles.
+	const { defaultEditorStyles, presetStyles } = useMemo( () => {
+		const _presetStyles =
+			editorSettings.styles?.filter(
+				( style ) =>
+					style.__unstableType && style.__unstableType !== 'theme'
+			) ?? [];
+		return {
+			presetStyles: _presetStyles,
+			defaultEditorStyles: [
+				...editorSettings.defaultEditorStyles,
+				..._presetStyles,
+			],
+		};
+	}, [ editorSettings.defaultEditorStyles, editorSettings.styles ] );
+
+	// Has theme styles if the theme supports them and if some styles were not preset styles (in which case they're theme styles).
+	const hasThemeStyles =
+		hasThemeStyleSupport &&
+		presetStyles.length !== ( editorSettings.styles?.length ?? 0 );
+
+	// If theme styles are not present or displayed, ensure that
+	// base layout styles are still present in the editor.
+	if ( ! editorSettings.disableLayoutStyles && ! hasThemeStyles ) {
+		defaultEditorStyles.push( {
```

Comment:
This `.push` call looks like a bad idea. The `defaultEditorStyles` array is memoized inside `useMemo`, and it is reused across multiple calls to `useEditorStyles`. But then each call of `useEditorStyles` pushes a new value to it, repeatedly. Then `defaultEditorStyles` ends up containing multiple instances of the pushed style.

Reply:
But before this PR, `defaultEditorStyles` was a local variable, constructed from scratch on each render. Now when it's memoized, it's a bug. Let's fix it before merging.